woocommerce/woocommerce-gutenberg-products-block
# Change default rows for product grid blocks to 3

Changes default rows for all grid blocks from 1 to 3. 3 feels like a better default, and specifically for the all products block it matches the 'grid' icon the block current has (3x3 grid of boxes).

_Note: For SSR blocks, existing blocks rows will change to 3, unless the user already changed them away from 1._ Gutenberg does not save default values to the block, so it's not possible to know if the user really set it to 1, or left the default enabled.

For the all products block we are handling deprecation, so existing blocks will not change. New blocks will be inserted with 3. 

**Additionally, all attributes, even defaults, will be saved to the block now. We're doing this using some filters.** This will prevent future issues if defaults are changed.

Fixes #1262

### How to test the changes in this Pull Request:

- Add all products block, or any "products by x" block and confirm 3 rows and 3 columns by default. Ensure you're using a default WP theme in case of overrides.
- Add a all products block using master (should have 1 row), save, then switch branches. Confirm there is still 1 row shown and there are no validation errors.
- Save a all products block. Check the post content source. Ensure all attributes are saved to the HTML comment.

### Changelog

> Changed default rows and columns for product grid blocks to 3x3.
